Source/Fuse.Elements/Resources/Exif/Exif.uno

```diff
@@ -0,0 +1,242 @@
+			else if defined(iOS)
+				_impl = new ExifIOSImpl(buffer);
+			else if defined(DOTNET)
+				_impl = new ExifDotNetImpl(buffer);
+		}
+
+		public ExifOrientation GetOrientation()
+		{
+			if (_impl == null)
+				return ExifOrientation.Undefined;
+			switch (_impl.GetOrientation())
+			{
+				case 1: return ExifOrientation.Normal;
+				case 2: return ExifOrientation.FlipHorizontal;
+				case 3: return ExifOrientation.Rotate180;
+				case 4: return ExifOrientation.FlipVertical;
+				case 5: return ExifOrientation.Transpose;
+				case 6: return ExifOrientation.Rotate90;
+				case 7: return ExifOrientation.Transverse;
+				case 8: return ExifOrientation.Rotate270;
+				default:
+					return ExifOrientation.Undefined;
+			}
+		}
+	}
+
+	[TargetSpecificImplementation, DotNetType("System.Drawing.Image")]
+	extern(CIL) class Image
+	{
+		[TargetSpecificImplementation]
+		public extern static Image FromStream(Stream stream);
+
+		[TargetSpecificImplementation]
+		public extern PropertyItem[] PropertyItems { get; }
+	}
+
+	[TargetSpecificImplementation, DotNetType("System.Drawing.Imaging.PropertyItem")]
+	extern(CIL) class PropertyItem
+	{
+		[TargetSpecificImplementation]
+		public extern int Id { get; set; }
+
+		[TargetSpecificImplementation]
+		public extern int Len { get; set; }
+
+		[TargetSpecificImplementation]
+		public extern short Type { get; set; }
+
+		[TargetSpecificImplementation]
+		public extern byte[] Value { get; set; }
+	}
+
+	extern(CIL) class ExifDotNetImpl : IExifImpl
+	{
+		Image _image;
+		public ExifDotNetImpl(byte[] bytes)
+		{
+			var memStream = new MemoryStream(bytes);
+			_image = Image.FromStream(memStream);
+		}
+
+		public int GetOrientation()
+		{
+			var properties = _image.PropertyItems;
+			foreach (var p in properties)
+			{
+				if (p.Id == Exif.OrientationTagId)
+				{
+					var b = new Buffer(p.Value);
+					var orientation = b.GetShort(0,true);
+					return orientation;
+				}
+			}
+			return 0;
+		}
+	}
+
+	[Require("Source.Include", "ImageIO/ImageIO.h")]
+	[Require("Source.Include", "CoreFoundation/CoreFoundation.h")]
+	[Require("Source.Include", "ExifHelper.h")]
+	[Require("Xcode.Framework", "ImageIO.framework")]
+	[Require("Xcode.Framework", "CoreFoundation.framework")]
+	extern(iOS) class ExifIOSImpl : IExifImpl
+	{
+		byte[] _bytes; 
+
+		public ExifIOSImpl(byte[] bytes)
+		{
+			_bytes = bytes;
+		}
+
+		byte ReadByte(int pos) {
+			return _bytes[pos];
+		}
+
+		public int GetOrientation()
+		{
+			return GetOrientationImpl(_bytes.Length, ReadByte);
+		}
+
+		[Foreign(Language.ObjC)]
+		int GetOrientationImpl(int length, Func<int,byte> readByte)
+		@{
+
+			CGDataProviderDirectCallbacks callbacks = {
+				.getBytePointer = 0,
+				.getBytesAtPosition = getBytesAtPosition
+			};
+			UnoCallbackBox box = { readByte };
+
+			CGDataProviderRef dataProvider = CGDataProviderCreateDirect(&box, length, &callbacks);
+			CGImageSourceRef source = CGImageSourceCreateWithDataProvider(dataProvider, NULL);
```

Comment:
You don't need the readByte-stuff, that's just overhead.
Here's a routine to create a CGImageSourceRef directly from a byte-array:
https://github.com/kusma/fuselibs-public/blob/bitmap/Source/Fuse.Common/Internal/Bitmap.uno#L110

Reply:
can't find this on the fusetools remote. Should i just copy the whole method?

Reply:
Yeah, it's not merged yet. You could just inline it into your function, if you want.